clarity/evaluator/haspi/eb.py

```diff
@@ -2034,48 +2028,35 @@ def ave_covary2(
     # Compute the time-frequency weights. The weight=1 if a segment in a
     # frequency band is above threshold, and weight=0 if below threshold.
     weight = np.zeros((n_channels, nseg))  # No IHC synchronization roll-off
-    wsync1 = np.zeros(
-        (n_channels, nseg)
-    )  # Loss of IHC synchronization at high frequencies
-    wsync2 = np.zeros((n_channels, nseg))
-    wsync3 = np.zeros((n_channels, nseg))
-    wsync4 = np.zeros((n_channels, nseg))
-    wsync5 = np.zeros((n_channels, nseg))
-    wsync6 = np.zeros((n_channels, nseg))
-    for k in range(n_channels):
-        for n in range(nseg):
-            if (
-                signal_rms[k, n] > threshold_db
-            ):  # Thresh in dB SL for including time-freq tile
-                weight[k, n] = 1
-                wsync1[k, n] = fsync[0, k]
-                wsync2[k, n] = fsync[1, k]
-                wsync3[k, n] = fsync[2, k]
-                wsync4[k, n] = fsync[3, k]
-                wsync5[k, n] = fsync[4, k]
-                wsync6[k, n] = fsync[5, k]
+    weight[signal_rms > threshold_db] = 1
+
+    # The wsync tensor should be constructed as follows:
+    #
+    # wsync = np.zeros((6, n_channels, nseg))
+    # for k in range(n_channels):
+    #    for n in range(nseg):
+    #        # Thresh in dB SL for including time-freq tile
+    #        if signal_rms[k, n] > threshold_db:
+    #            wsync[:, k, n] = fsync[:, k]
+    #
+    # This can be written is an efficient vectorsized form as follows:
+    wsync = np.zeros((6, n_channels, nseg))
+    mask = signal_rms > threshold_db
+    fsync3d = np.repeat(fsync[..., None], nseg, axis=2)
+    wsync[:, mask] = fsync3d[:, mask]
```

Comment:
We should try to do this in more parts of the eb.
For sure there are many places we can improve.

Reply:
After this is merge, I will check if I can find ways to improve HAAQI